src/components/badgeable/Badgeable.js

```diff
@@ -18,11 +18,19 @@ type Props = {
 };
 
 const Badgeable = (props: Props) => {
-    const { children, className = '', topLeft = null, topRight = null, bottomLeft = null, bottomRight = null } = props;
+    const {
+        children,
+        className = '',
+        topLeft = null,
+        topRight = null,
+        bottomLeft = null,
+        bottomRight = null,
+        ...rest
+    } = props;
 
     return (
         <div className={`badgeable-container ${className}`}>
-            {children}
+            {React.isValidElement(children) ? React.cloneElement(React.Children.only(children), rest) : children}
```

Comment:
What is the purpose of this change? I would think that since `children` should always be declarative (eslint rule), props could be passed directly rather than via `Badgeable`.

Reply:
We have a pattern in EUA of using `HeaderToggleButton` to wrap a button and pass a `color` prop depending on the state of the header and the theme. For NotificationCenterButton and TaskFlyout we wrap these buttons in a `Badgeable` as well. Due to this, there's currently a bug where the the inner icon element is not receiving the correct color prop from `HeaderToggleButton`, and this would fix that issue.

Reply:
It sounds like the `HeaderToggleButton` component should be improved to allow arbitrary content to be rendered rather than decorated, if that's the case. Would it be possible to add a render prop to `HeaderToggleButton` so you can access the `color` prop?